src/test/java/org/myrobotlab/codec/CodecUtilsTest.java

```diff
@@ -24,27 +25,106 @@ public void testLocale() {
+        if (!CodecUtils.USING_GSON) {
+            assertEquals("{\"language\":null,\"displayLanguage\":null,\"country\":null,\"displayCountry\":null,\"tag\":null,\"class\":\"org.myrobotlab.service.data.Locale\"}", json);
+        } else {
+            assertEquals("{\"class\":\"org.myrobotlab.service.data.Locale\"}", json);
+        }
 
         code = "";
         mrlLocale = new Locale(code);
         json = CodecUtils.toJson(mrlLocale);
-        assertEquals("{\"language\":null,\"displayLanguage\":null,\"country\":null,\"displayCountry\":null,\"tag\":null,\"class\":\"org.myrobotlab.service.data.Locale\"}", json);
+        if (!CodecUtils.USING_GSON) {
+            assertEquals("{\"language\":null,\"displayLanguage\":null,\"country\":null,\"displayCountry\":null,\"tag\":null,\"class\":\"org.myrobotlab.service.data.Locale\"}", json);
+        } else {
+            assertEquals("{\"class\":\"org.myrobotlab.service.data.Locale\"}", json);
+        }
 
         code = "-uS";
         mrlLocale = new Locale(code);
         json = CodecUtils.toJson(mrlLocale);
         assertEquals("{\"language\":\"\",\"displayLanguage\":\"\",\"country\":\"US\",\"displayCountry\":\"United States\",\"tag\":\"-US\",\"class\":\"org.myrobotlab.service.data.Locale\"}", json);
 
     }
+
+    @Test
+    public void testJsonMessageSerDeser() {
+        Message message = Message.createMessage(
+                "testing-sender",
+                "testService",
+                "testMethod",
+                        new Object[]{
+                                "testParam1",
+                                2,
+                                new MRLListener(
+                                        "topic",
+                                        "callbackService",
+                                        "callbackMethod"
+                                )
+                        }
+                );
+        String json = CodecUtils.toJsonMsg(message);
+        Message deserMessage = CodecUtils.jsonToMessage(json);
+        assertEquals(message, deserMessage);
+    }
+
+    @Test
+    public void testDecodeMessageParams() {
+        Message encodedMessage = Message.createMessage(
+                "testing-sender",
+                "testService",
+                "testMethod",
+                new Object[]{
+                        "\"testParam1\"",
+                        "2",
+                        CodecUtils.toJson(new MRLListener(
+                                "topic",
+                                "callbackService",
+                                "callbackMethod"
+                        ))
+                }
+        );
+
+        Message decodedMessage = Message.createMessage(
+                "testing-sender",
+                "testService",
+                "testMethod",
+                new Object[]{
+                        "testParam1",
+                        2,
```

Comment:
heh.. it be nice to see all primitive types and a few null values :)

Reply:
Added some more types and a list and nested list. In the process found a bug with how GSON behaves so it's good that I added those